ext/curl/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+zephyr_library()
+zephyr_library_include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include/curl)
+target_include_directories(app PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include/curl)
+target_include_directories(app PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/lib)
```

Comment:
The `app` lib is for the application, and should be under control of the user.

We should never manipulate this library directly in Zephyr build system.
If include paths are needed from a library in the app, then please create an interface lib, and a corresponding `APP_LINK_WITH_<lib>` Kconfig setting.

For inspiration: https://github.com/zephyrproject-rtos/zephyr/blob/94e8c2eb81f390a9226a3b70fdb69a002642f274/modules/mbedtls/CMakeLists.txt#L9-L11
https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/mbedtls/Kconfig#L154-L160

This will allow a user to create another lib (not `app`) and then link that lib to the interface, and not being forced into placing everything in `app`.

Reply:
This shouldn't be necessary as we want curl to be a normal library like iperf3 and getopt.

Reply:
Used this approach.